app/questionnaire/router.py

```diff
@@ -413,6 +414,7 @@ def _get_return_url_for_inaccessible_location(
         if (
             not is_for_previous
             and return_to_block_id
+            and routing_path.section_id in self.enabled_section_ids
```

Comment:
It would be for anything that returns to a different section that might be disabled, which I think at the moment is only GCS, so happy to move the check